sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -1246,3 +1246,13 @@ void *coap_request_ctx_get_context(const coap_request_ctx_t *ctx)
 {
     return ctx->context;
 }
+
+uint32_t coap_request_ctx_get_tl_type(const coap_request_ctx_t *ctx)
+{
+#ifdef MODULE_GCOAP
+    return ctx->tl_type;
+#else
+    (void)ctx;
+    return 0;
+#endif
+}
```

Comment:
As `tl_type` is purely a `gcoap` thing: why not move this function to `gcoap.c` as `gcoap_request_ctx_get_tl_type()`?

Reply:
The problem there is that `nanocoap_internal.h` can only be accessed from the `nanocoap` module since it's not a global include. 

> `tl_type` is purely a `gcoap` thing

Well it doesn't have to stay like that :wink:

Reply:
> The problem there is that `nanocoap_internal.h` can only be accessed from the `nanocoap` module since it's not a global include.

Mhh... if it is internal to `nanocoap`, how will that then serve in a `gcoap` response handler?

Reply:
> > `tl_type` is purely a `gcoap` thing
> 
> Well it doesn't have to stay like that wink

then why guard it with `MODULE_GCOAP`?

Reply:
> if it is internal to `nanocoap`, how will that then serve in a `gcoap` response handler?

The accessor function is in `net/nanocoap.h`